src/Suggest/Phrase.php

```diff
@@ -138,30 +139,18 @@ public function setSmoothingModel(string $model, array $params): Phrase
     /**
      * @return $this
      */
-    public function addCandidateGenerator(AbstractCandidateGenerator $generator): Phrase
+    public function addDirectGenerator(DirectGenerator $generator): Phrase
     {
-        return $this->setParam('candidate_generator', $generator);
+        return $this->addParam('direct_generator', $generator);
     }
 
     /**
-     * {@inheritdoc}
+     * @deprecated Please use addDirectGenerator instead.
+     *
+     * @return $this
      */
-    public function toArray(): array
+    public function addCandidateGenerator(AbstractCandidateGenerator $generator): Phrase
     {
-        $array = parent::toArray();
-
-        $baseName = $this->_getBaseName();
-
-        if (isset($array[$baseName]['candidate_generator'])) {
-            $generator = $array[$baseName]['candidate_generator'];
-            unset($array[$baseName]['candidate_generator']);
-
-            $keys = \array_keys($generator);
-            $values = \array_values($generator);
-
-            $array[$baseName][$keys[0]][] = $values[0];
-        }
-
-        return $array;
+        return $this->addParam('candidate_generator', $generator);
```

Comment:
```suggestion
        \trigger_deprecation('ruflin/elastica', '7.2.0', 'The "%s()" method is deprecated, use the "addDirectGenerator()" method instead. It will be removed in 8.0.', __METHOD__);

        return $this->addParam('candidate_generator', $generator);
```